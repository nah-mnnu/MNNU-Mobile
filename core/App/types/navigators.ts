import { NavigatorScreenParams } from '@react-navigation/core'

import { DeclineType } from './decline'

export enum Screens {
  AttemptLockout = 'Temporarily Locked',
  Splash = 'Splash',
  Onboarding = 'Onboarding',
  Terms = 'Terms',
  CreatePIN = 'Create a PIN',
  EnterPIN = 'Enter PIN',
  Home = 'Home',
  Scan = 'Scan',
  Credentials = 'Credentials',
  CredentialDetails = 'Credential Details',
  Notifications = 'Notifications',
  CredentialOffer = 'Credential Offer',
  ProofRequest = 'Proof Request',
  ProofRequestAttributeDetails = 'Proof Request Attribute Details',
  Settings = 'Settings',
  Language = 'Language',
  Contacts = 'Contacts',
  ContactDetails = 'Contact Details',
  Chat = 'Chat',
  Connection = 'Connection',
  OnTheWay = 'On The Way',
  Declined = 'Declined',
  CommonDecline = 'Common Decline',
  UseBiometry = 'Use Biometry',
  RecreatePIN = 'Change PIN',
  Developer = 'Developer',
}

export enum Stacks {
  TabStack = 'Tab Stack',
  HomeStack = 'Home Stack',
  ConnectStack = 'Connect Stack',
  CredentialStack = 'Credentials Stack',
  SettingStack = 'Settings Stack',
  ContactStack = 'Contacts Stack',
  NotificationStack = 'Notifications Stack',
  ConnectionStack = 'Connection Stack',
}

export enum TabStacks {
  HomeStack = 'Tab Home Stack',
  ConnectStack = 'Tab Connect Stack',
  CredentialStack = 'Tab Credential Stack',
}

export type RootStackParams = {
  [Screens.Splash]: undefined
  [Stacks.TabStack]: NavigatorScreenParams<TabStackParams>
  [Stacks.ConnectStack]: NavigatorScreenParams<ConnectStackParams>
  [Stacks.SettingStack]: NavigatorScreenParams<SettingStackParams>
  [Stacks.ContactStack]: NavigatorScreenParams<ContactStackParams>
  [Stacks.NotificationStack]: NavigatorScreenParams<NotificationStackParams>
}

export type TabStackParams = {
  [TabStacks.HomeStack]: NavigatorScreenParams<HomeStackParams>
  [TabStacks.ConnectStack]: NavigatorScreenParams<ConnectStackParams>
  [TabStacks.CredentialStack]: NavigatorScreenParams<CredentialStackParams>
}

export type AuthenticateStackParams = {
  [Screens.Onboarding]: undefined
  [Screens.Terms]: undefined
  [Screens.AttemptLockout]: undefined
  [Screens.CreatePIN]: { setAuthenticated: (status: boolean) => void } | undefined
  [Screens.EnterPIN]: { setAuthenticated: (status: boolean) => void } | undefined
  [Screens.UseBiometry]: undefined
}

export type ContactStackParams = {
  [Screens.Contacts]: undefined
  [Screens.Chat]: { connectionId: string }
  [Screens.ContactDetails]: { connectionId: string }
}

export type CredentialStackParams = {
  [Screens.Credentials]: undefined
  [Screens.CredentialDetails]: { credentialId: string }
}

export type HomeStackParams = {
  [Screens.Home]: undefined
  [Screens.Notifications]: undefined
}

export type ConnectStackParams = {
  [Screens.Scan]: undefined
}

export type SettingStackParams = {
  [Screens.Settings]: undefined
  [Screens.Language]: undefined
  [Screens.UseBiometry]: undefined
  [Screens.CreatePIN]: undefined
  [Screens.RecreatePIN]: undefined
  [Screens.Terms]: undefined
  [Screens.Onboarding]: undefined
  [Screens.Developer]: undefined
}

export type NotificationStackParams = {
  [Screens.CredentialDetails]: { credentialId: string }
  [Screens.CredentialOffer]: { credentialId: string }
  [Screens.ProofRequest]: { proofId: string }
  [Screens.ProofRequestAttributeDetails]: {
    proofId: string
    attributeName: string | null
  }
  [Screens.CommonDecline]: {
    declineType: DeclineType
    itemId: string
  }
}

export type DeliveryStackParams = {
  [Screens.Connection]: { connectionId?: string; threadId?: string }
  [Screens.CredentialOffer]: { credentialId: string }
  [Screens.ProofRequest]: { proofId: string }
  [Screens.ProofRequestAttributeDetails]: {
    proofId: string
    attributeName: string | null
  }
  [Screens.CommonDecline]: {
    declineType: DeclineType
    itemId: string
  }
  [Screens.OnTheWay]: { credentialId: string }
  [Screens.Declined]: { credentialId: string }
}
