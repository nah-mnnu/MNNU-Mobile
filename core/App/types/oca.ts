import { CredentialExchangeRecord, CredentialMetadataKeys } from '@aries-framework/core'

import { parsedCredDefName } from '../utils/cred-def'
import { hashCode, hashToRGBA } from '../utils/helpers'

import { Attribute, Field } from './record'

export enum BaseType {
  BINARY = 'Binary',
  TEXT = 'Text',
  DATETIME = 'DateTime',
  NUMERIC = 'Numeric',
  DATEINT = 'DateInt',
}

export enum OverlayType {
  BASE_10 = 'spec/capture_base/1.0',
  META_10 = 'spec/overlays/meta/1.0',
  LABEL_10 = 'spec/overlays/label/1.0',
  CARD_LAYOUT_10 = 'spec/overlays/card_layout/1.0',
  FORMAT_10 = 'spec/overlays/format/1.0',
  ENCODING_10 = 'spec/overlays/character_encoding/1.0',
}

export interface Bundle {
  capture_base: CaptureBaseOverlay
  overlays: BaseOverlay[]
}

export interface Bundles {
  [key: string]: Bundle | string
}

export interface BaseOverlay {
  type: string
  capture_base: string
}

export interface BaseL10nOverlay extends BaseOverlay {
  language: string
}

export interface AttributeLabels {
  [key: string]: string
}

export interface LabelOverlay extends BaseL10nOverlay {
  attr_labels: AttributeLabels
}

export interface FormatOverlay extends BaseL10nOverlay {
  attr_formats: AttributeLabels
}

export interface CharacterEncodingOverlay extends BaseL10nOverlay {
  attr_character_encoding: AttributeLabels
}

export interface CaptureBaseOverlay extends BaseOverlay {
  type: string
  attributes?: AttributeLabels
}

export interface MetaOverlay extends BaseL10nOverlay {
  name: string
  issuerName?: string
}

export interface CardLayoutOverlay extends BaseOverlay {
  backgroundColor?: string
  imageSource?: string
  header?: OverlayHeader
  footer?: OverlayFooter
}
export interface OverlayHeader {
  color?: string
  backgroundColor?: string
  imageSource?: string
  hideIssuer?: boolean
}

export interface OverlayFooter {
  color?: string
  backgroundColor?: string
}

export interface OCACredentialBundle {
  getCaptureBase(): CaptureBaseOverlay
  getOverlay<F extends BaseOverlay>(type: string, language?: string): F | undefined
  getLabelOverlay(language: string): LabelOverlay | undefined
  getFormatOverlay(): FormatOverlay | undefined
  getCharacterEncodingOverlay(): CharacterEncodingOverlay | undefined
  getCardLayoutOverlay(): CardLayoutOverlay | undefined
  getMetaOverlay(language: string): MetaOverlay | undefined
}
export interface OCABundleResolver {
  getCredentialPresentationFields(credential: CredentialExchangeRecord, language: string): Promise<Field[]>
  loadDefaultBundles(): OCABundleResolver
  loadBundles(bundles: Bundles): OCABundleResolver
  resolve(credential: CredentialExchangeRecord): Promise<OCACredentialBundle | undefined>
  resolveDefaultBundle(credential: CredentialExchangeRecord): Promise<OCACredentialBundle | undefined>
}
export class DefaultOCACredentialBundle implements OCACredentialBundle {
  private bundle: Bundle
  public constructor(bundle: Bundle) {
    this.bundle = bundle
  }
  public getCardLayoutOverlay(): CardLayoutOverlay | undefined {
    const layout = this.getOverlay<CardLayoutOverlay>(OverlayType.CARD_LAYOUT_10)
    return layout
  }
  public getMetaOverlay(language: string): MetaOverlay | undefined {
    return this.getOverlay<MetaOverlay>(OverlayType.META_10, language)
  }
  public getCaptureBase(): CaptureBaseOverlay {
    const overlay = this.getOverlay<CaptureBaseOverlay>(OverlayType.BASE_10)
    if (overlay === undefined) {
      throw new Error('Expected Capture Base to be defined')
    }
    return overlay
  }
  public getLabelOverlay(language: string): LabelOverlay | undefined {
    return this.getOverlay<LabelOverlay>(OverlayType.LABEL_10, language)
  }
  public getFormatOverlay(): FormatOverlay | undefined {
    return this.getOverlay<FormatOverlay>(OverlayType.FORMAT_10)
  }
  public getCharacterEncodingOverlay(): CharacterEncodingOverlay | undefined {
    return this.getOverlay<CharacterEncodingOverlay>(OverlayType.ENCODING_10)
  }
  public getOverlay<F extends BaseOverlay>(type: string, language?: string): F | undefined {
    if (type === OverlayType.BASE_10) {
      return (this.bundle as Bundle).capture_base as unknown as F
    }
    if (language !== undefined) {
      return (this.bundle as Bundle).overlays.find(
        (item) => item.type === type.toString() && (item as BaseL10nOverlay).language === language
      ) as F
    }
    return (this.bundle as Bundle).overlays.find((item) => item.type === type.toString()) as F
  }
}

export class DefaultOCABundleResolver implements OCABundleResolver {
  private bundles: Bundles = {}
  public resolve(credential: CredentialExchangeRecord): Promise<OCACredentialBundle | undefined> {
    const credentialDefinitionId = credential.metadata.get(
      CredentialMetadataKeys.IndyCredential
    )?.credentialDefinitionId
    const schemaId = credential.metadata.get(CredentialMetadataKeys.IndyCredential)?.schemaId
    for (const item of [credentialDefinitionId, schemaId]) {
      if (item && this.bundles[item] !== undefined) {
        let bundle = this.bundles[item]
        // if it is a string, it is a reference/alias to another one bundle
        if (typeof bundle === 'string') {
          bundle = this.bundles[bundle]
        }
        return Promise.resolve(new DefaultOCACredentialBundle(bundle as Bundle))
      }
    }
    return Promise.resolve(undefined)
  }
  public resolveDefaultBundle(credential: CredentialExchangeRecord): Promise<OCACredentialBundle | undefined> {
    const defaultMetaOverlay: MetaOverlay = {
      capture_base: '',
      type: OverlayType.META_10,
      name: parsedCredDefName(credential),
      language: 'en',
      issuerName: credential?.connectionId ?? '',
    }
    const defaultCardLayoutLayer: CardLayoutOverlay = {
      capture_base: '',
      type: OverlayType.CARD_LAYOUT_10,
      backgroundColor: hashToRGBA(hashCode(defaultMetaOverlay?.name ?? '')),
    }
    const bundle: Bundle = {
      capture_base: { capture_base: '', type: OverlayType.BASE_10 },
      overlays: [defaultMetaOverlay, defaultCardLayoutLayer],
    }
    return Promise.resolve(new DefaultOCACredentialBundle(bundle))
  }
  public loadBundles(bundles: Bundles): OCABundleResolver {
    Object.assign(this.bundles, bundles)
    return this
  }
  public loadDefaultBundles(): OCABundleResolver {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    return this.loadBundles(require('../assets/oca-bundles.json'))
  }
  public async getCredentialPresentationFields(
    credential: CredentialExchangeRecord,
    language: string
  ): Promise<Array<Field>> {
    const bundle = await this.resolve(credential)
    const baseOverlay = bundle?.getCaptureBase()
    const labelOverlay = bundle?.getLabelOverlay(language)
    const formatOverlay = bundle?.getFormatOverlay()
    const characterEncodingOverlay = bundle?.getCharacterEncodingOverlay()
    const _fields: Array<Field> = []
    if (baseOverlay && baseOverlay.attributes) {
      for (const key in baseOverlay?.attributes) {
        if (Object.prototype.hasOwnProperty.call(baseOverlay?.attributes, key)) {
          const _sourceField = credential?.credentialAttributes?.find((item) => item.name === key)
          if (_sourceField) {
            const _field: Field = {
              name: _sourceField.name,
              value: _sourceField.value,
              mimeType: _sourceField.mimeType,
            } as Attribute
            if (labelOverlay?.attr_labels[key]) {
              _field.label = labelOverlay?.attr_labels[key]
            }
            _field.format = formatOverlay?.attr_formats[key]
            _field.type = baseOverlay?.attributes[key]
            _field.encoding = characterEncodingOverlay?.attr_character_encoding[key]

            _fields.push(_field)
          }
        }
      }
    } else {
      if (credential.credentialAttributes) {
        for (const _sourceField of credential.credentialAttributes) {
          const _field: Field = {
            name: _sourceField.name,
            value: _sourceField.value,
            mimeType: _sourceField.mimeType,
          } as Attribute
          if (labelOverlay?.attr_labels[_sourceField.name]) {
            _field.label = labelOverlay?.attr_labels[_sourceField.name]
          }
          _field.format = formatOverlay?.attr_formats[_sourceField.name]
          _field.type = baseOverlay?.attributes?.[_sourceField.name]
          _field.encoding = characterEncodingOverlay?.attr_character_encoding[_sourceField.name]

          _fields.push(_field)
        }
      }
    }
    return _fields
  }
}
