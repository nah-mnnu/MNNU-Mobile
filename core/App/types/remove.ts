export enum RemoveType {
  Credential = 'Credential',
  Contact = 'Contact',
}
