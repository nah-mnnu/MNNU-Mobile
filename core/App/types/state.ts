import { BifoldError } from './error'

export interface Onboarding {
  didCompleteTutorial: boolean
  didAgreeToTerms: boolean
  didCreatePIN: boolean
  didConsiderBiometry: boolean
}

export interface Preferences {
  useBiometry: boolean
  developerModeEnabled: boolean
}

export interface Lockout {
  displayNotification: boolean
}

export interface LoginAttempt {
  lockoutDate?: number
  servedPenalty: boolean
  loginAttempts: number
}

export interface Privacy {
  didShowCameraDisclosure: boolean
}

export interface Authentication {
  didAuthenticate: boolean
}

export interface State {
  onboarding: Onboarding
  authentication: Authentication
  privacy: Privacy
  lockout: Lockout
  loginAttempt: LoginAttempt
  preferences: Preferences
  error: BifoldError | null
  loading: boolean
}
