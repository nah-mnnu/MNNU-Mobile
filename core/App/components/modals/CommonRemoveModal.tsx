import * as React from 'react'
import { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { Modal, ScrollView, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import Collapsible from 'react-native-collapsible'
import { SafeAreaView } from 'react-native-safe-area-context'
import Icon from 'react-native-vector-icons/MaterialIcons'

import { useTheme } from '../../contexts/theme'
import { GenericFn } from '../../types/fn'
import { RemoveType } from '../../types/remove'
import { testIdWithKey } from '../../utils/testable'
import Button, { ButtonType } from '../buttons/Button'
import UnorderedList from '../misc/UnorderedList'
import FauxNavigationBar from '../views/FauxNavigationBar'

interface CommonRemoveModalProps {
  removeType: RemoveType
  onSubmit?: GenericFn
  onCancel?: GenericFn
  visible?: boolean
}

interface RemoveProps {
  title: string
  content: string[]
}

const Dropdown: React.FC<RemoveProps> = ({ title, content }) => {
  const { TextTheme, ColorPallet } = useTheme()
  const [isCollapsed, setIsCollapsed] = useState<boolean>(true)

  return (
    <>
      <TouchableOpacity
        onPress={() => setIsCollapsed(!isCollapsed)}
        style={[
          {
            padding: 15,
            backgroundColor: ColorPallet.brand.modalSecondaryBackground,
            borderRadius: 5,
            flexDirection: 'row',
            justifyContent: 'space-between',
          },
        ]}
      >
        <Text style={[TextTheme.normal, { fontWeight: 'bold' }]}>{title}</Text>
        <Icon name={isCollapsed ? 'expand-more' : 'expand-less'} size={24} color={TextTheme.normal.color} />
      </TouchableOpacity>
      <Collapsible collapsed={isCollapsed} enablePointerEvents={true}>
        <View
          style={[{ marginTop: 10, borderLeftWidth: 2, borderLeftColor: ColorPallet.brand.modalSecondaryBackground }]}
        >
          <UnorderedList UnorderedListItems={content} />
        </View>
      </Collapsible>
    </>
  )
}

const CommonRemoveModal: React.FC<CommonRemoveModalProps> = ({ removeType, visible, onSubmit, onCancel }) => {
  if (!removeType) {
    throw new Error('removeType cannot be undefined')
  }

  const { t } = useTranslation()
  const { ColorPallet, TextTheme } = useTheme()

  const styles = StyleSheet.create({
    container: {
      height: '100%',
      backgroundColor: ColorPallet.brand.modalPrimaryBackground,
      padding: 20,
    },
    controlsContainer: {
      marginTop: 'auto',
      marginHorizontal: 20,
      marginBottom: 108,
    },
  })

  return (
    <Modal visible={visible} animationType="slide">
      <FauxNavigationBar title={t('CredentialDetails.RemoveFromWallet')} />
      <SafeAreaView
        edges={['left', 'right', 'bottom']}
        style={{ backgroundColor: ColorPallet.brand.modalPrimaryBackground }}
      >
        <ScrollView style={[styles.container]}>
          {removeType === RemoveType.Contact && (
            <View>
              <View style={[{ marginBottom: 25 }]}>
                <Text style={[TextTheme.title]}>{t('ContactDetails.RemoveTitle')}</Text>
              </View>
              <View>
                <Text style={[TextTheme.normal]}>{t('ContactDetails.RemoveCaption')}</Text>
              </View>
            </View>
          )}
          {removeType === RemoveType.Credential && (
            <View>
              <View style={[{ marginBottom: 25 }]}>
                <Text style={[TextTheme.title]}>{t('CredentialDetails.RemoveTitle')}</Text>
              </View>
              <View>
                <Text style={[TextTheme.normal]}>{t('CredentialDetails.RemoveCaption')}</Text>
              </View>
              <View style={{ marginTop: 25 }}>
                <Dropdown
                  title={t('CredentialDetails.YouWillNotLose')}
                  content={[
                    t('CredentialDetails.YouWillNotLoseListItem1'),
                    t('CredentialDetails.YouWillNotLoseListItem2'),
                  ]}
                />
              </View>
              <View style={{ marginTop: 25 }}>
                <Dropdown
                  title={t('CredentialDetails.HowToGetThisCredentialBack')}
                  content={[t('CredentialDetails.HowToGetThisCredentialBackListItem1')]}
                />
              </View>
            </View>
          )}
        </ScrollView>
        <View style={[styles.controlsContainer]}>
          <View style={[{ paddingTop: 10 }]}>
            <Button
              title={t('CredentialDetails.RemoveFromWallet')}
              accessibilityLabel={t('CredentialDetails.RemoveFromWallet')}
              testID={testIdWithKey('ConfirmRemoveButton')}
              onPress={() => onSubmit && onSubmit()}
              buttonType={ButtonType.Critical}
            />
          </View>
          <View style={[{ paddingTop: 10 }]}>
            <Button
              title={t('Global.Cancel')}
              accessibilityLabel={t('Global.Cancel')}
              testID={testIdWithKey('AbortRemoveButton')}
              onPress={() => onCancel && onCancel()}
              buttonType={ButtonType.Secondary}
            />
          </View>
        </View>
      </SafeAreaView>
    </Modal>
  )
}

export default CommonRemoveModal
