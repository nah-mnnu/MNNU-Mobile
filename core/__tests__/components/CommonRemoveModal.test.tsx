import { render } from '@testing-library/react-native'
import React from 'react'

import CommonRemoveModal from '../../App/components/modals/CommonRemoveModal'
import { RemoveType } from '../../App/types/remove'

describe('ErrorModal Component', () => {
  test('Renders correctly', async () => {
    const tree = render(<CommonRemoveModal removeType={RemoveType.Credential} />)

    expect(tree).toMatchSnapshot()
  })
})
