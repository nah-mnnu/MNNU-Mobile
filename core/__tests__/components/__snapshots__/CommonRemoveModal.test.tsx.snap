// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal Component Renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#42803E",
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "center",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#42803E",
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "flexGrow": 1,
            "fontSize": 17,
            "fontWeight": "bold",
            "paddingBottom": 12,
            "textAlign": "center",
          }
        }
      >
        CredentialDetails.RemoveFromWallet
      </Text>
    </View>
  </View>
  <RNCSafeAreaView
    edges={
      Array [
        "left",
        "right",
        "bottom",
      ]
    }
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    <RCTScrollView
      style={
        Array [
          Object {
            "backgroundColor": "#000000",
            "height": "100%",
            "padding": 20,
          },
        ]
      }
    >
      <View>
        <View>
          <View
            style={
              Array [
                Object {
                  "marginBottom": 25,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              CredentialDetails.RemoveTitle
            </Text>
          </View>
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              CredentialDetails.RemoveCaption
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 25,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#313132",
                  "borderRadius": 5,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 18,
                      "fontWeight": "normal",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                  ]
                }
              >
                CredentialDetails.YouWillNotLose
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 24,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="auto"
              style={
                Object {
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                nativeID="animatedComponent"
                onLayout={[Function]}
                style={Object {}}
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderLeftColor": "#313132",
                        "borderLeftWidth": 2,
                        "marginTop": 10,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "display": "flex",
                          "flexDirection": "row",
                          "marginBottom": 5,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#FFFFFF",
                            "fontSize": 18,
                            "fontWeight": "normal",
                          },
                          Object {
                            "paddingLeft": 5,
                          },
                        ]
                      }
                    >
                      •
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#FFFFFF",
                            "fontSize": 18,
                            "fontWeight": "normal",
                          },
                          Object {
                            "paddingLeft": 5,
                          },
                        ]
                      }
                    >
                      CredentialDetails.YouWillNotLoseListItem1
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "display": "flex",
                          "flexDirection": "row",
                          "marginBottom": 5,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#FFFFFF",
                            "fontSize": 18,
                            "fontWeight": "normal",
                          },
                          Object {
                            "paddingLeft": 5,
                          },
                        ]
                      }
                    >
                      •
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#FFFFFF",
                            "fontSize": 18,
                            "fontWeight": "normal",
                          },
                          Object {
                            "paddingLeft": 5,
                          },
                        ]
                      }
                    >
                      CredentialDetails.YouWillNotLoseListItem2
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "marginTop": 25,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#313132",
                  "borderRadius": 5,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 18,
                      "fontWeight": "normal",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                  ]
                }
              >
                CredentialDetails.HowToGetThisCredentialBack
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 24,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="auto"
              style={
                Object {
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                nativeID="animatedComponent"
                onLayout={[Function]}
                style={Object {}}
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderLeftColor": "#313132",
                        "borderLeftWidth": 2,
                        "marginTop": 10,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "display": "flex",
                          "flexDirection": "row",
                          "marginBottom": 5,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#FFFFFF",
                            "fontSize": 18,
                            "fontWeight": "normal",
                          },
                          Object {
                            "paddingLeft": 5,
                          },
                        ]
                      }
                    >
                      •
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#FFFFFF",
                            "fontSize": 18,
                            "fontWeight": "normal",
                          },
                          Object {
                            "paddingLeft": 5,
                          },
                        ]
                      }
                    >
                      CredentialDetails.HowToGetThisCredentialBackListItem1
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "marginBottom": 108,
            "marginHorizontal": 20,
            "marginTop": "auto",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingTop": 10,
            },
          ]
        }
      >
        <View
          accessibilityLabel="CredentialDetails.RemoveFromWallet"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#42803E",
              "borderRadius": 4,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/ConfirmRemoveButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              CredentialDetails.RemoveFromWallet
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingTop": 10,
            },
          ]
        }
      >
        <View
          accessibilityLabel="Global.Cancel"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "#42803E",
              "borderRadius": 4,
              "borderWidth": 2,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/AbortRemoveButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#42803E",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Global.Cancel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;
